### PR TITLE
Tracing without performance

### LINKED ISSUE #1541: Tracing without Performance
See Project: https://github.com/getsentry/team-sdks/issues/5
See Project Board: [github.com/orgs/getsentry/projects/110](https://github.com/orgs/getsentry/projects/110)

Note: implementation details/requirements for Mobile (vs. backend) SDKs TBD. PR for Java can act as reference: https://github.com/getsentry/sentry-java/pull/2788

### PR DESCRIPTION
**Pull request by buenaflor**
## :scroll: Description
Implement tracing without performance, more context in the referenced issue.

## :bulb: Motivation and Context
Closes https://github.com/getsentry/sentry-dart/issues/1541

## :green_heart: How did you test it?
Unit tests

## :pencil: Checklist

- [ ] I reviewed submitted code
- [ ] I added tests to verify changes
- [ ] No new PII added or SDK only sends newly added PII if `sendDefaultPii` is enabled
- [ ] I updated the docs if needed
- [ ] All tests passing
- [ ] No breaking changes

### CODE REVIEW

**Comment by buenaflor** [comment]
This is a preview of the implementation, I don't know exactly if this is correct other than the unit tests. Please lmk if something is missing or doesn't look like it's right.

**Comment by markushi** [comment]
Looking good! I think there's one missing part though: New trace headers need to be generated whenever a new transaction is started, but performance is disabled. 

Basically turning existing constructs like
```java
if (performanceEnabled) {
  startNewTransaction();
}
```
into 
```java
if (performanceEnabled) {
  startNewTransaction();
} else {
  startNewTrace();
}
```
E.g. see this one: https://github.com/getsentry/sentry-java/blob/9e60fc1fe6e17eaf16620e62eff24459aeec2505/sentry-android-core/src/main/java/io/sentry/android/core/ActivityLifecycleIntegration.java#L178-L184

**Comment by brustolin** [comment]
> Looking good! I think there's one missing part though: New trace headers need to be generated whenever a new transaction is started, but performance is disabled.

I don't think we should create a new trace header every time, but one for the entire session.

Also, this is missing the propagation of the trace for http requests. Are you planning to create a new PR for this?

**Comment by buenaflor** [comment]
I've added the propagation for http requests, pls lmk if those make sense

**Comment by brustolin** [comment]
> I've added the propagation for http requests, pls lmk if those make sense

Yeap. Looks good.

**Review by brustolin — APPROVED** [review]
LGTM

**Comment by kahest** [comment]
@denrase if you have some time maybe good if you had a look at this to be in the loop :)

File: dart/lib/src/sentry_baggage.dart [hunk 1]
```diff
@@ -87,6 +90,27 @@ class SentryBaggage {
     return SentryBaggage(keyValues, logger: logger);
   }
 
+  @internal
+  setValuesFromScope(Scope scope, SentryOptions options) {
```
Comment (denrase) [inline dart/lib/src/sentry_baggage.dart:94]: On Java we are setting `transaction` and `sampleRate` to null. Is this also needed here?

File: dart/lib/src/propagation_context.dart [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import 'package:meta/meta.dart';
+
+import '../sentry.dart';
+
+@internal
+class PropagationContext {
+  late SentryId traceId;
```
Comment (denrase) [inline dart/lib/src/propagation_context.dart]: Could we just assign the random Ids when the properties are declared? No need for a explicit constructor then.

```
@internal
class PropagationContext {
  SentryId traceId = SentryId.newId();
  SpanId spanId = SpanId.newId();
  
...

}
```

**Review by denrase — COMMENTED** [review]
Just some general questions & suggestions, as i'm lacking context regarding the overall 'Tracing without performace' issue.

**Pull request merged**